tests/src/test/scala/akka/kafka/scaladsl/RebalanceSpec.scala

```diff
@@ -243,6 +250,212 @@ class RebalanceSpec extends SpecBase with TestcontainersKafkaLike with Inside {
+      var t2 = subscribeAndConsumeMessages(consumerClientId2, probe2subscription, topicCount, partitionCount)
+      var (control2, probe2) = (t2._1, t2._2)
+      probe2.ensureSubscription()
+      probe2.request(1)
+      log.debug(s"END:2:Subscribe client ${consumerClientId2} to all topic-partitions in parallel")
+      sleep(rebalanceEventBufferTime, s"SLEEP:2:sleep to allow consume messages by ${consumerClientId2}")
+      // END: introduce second consumer2 with topic-partitions divided between two consumers
+
+      // BEGIN: introduce third consumer3 with all topic-partitions divided into three consumers
+      log.debug(s"BEGIN:3:Subscribe client ${consumerClientId3} to all topic-partitions in parallel")
+      val probe3rebalanceActor = TestProbe()
+      val probe3subscription = Subscriptions.topics(topicSet).withRebalanceListener(probe3rebalanceActor.ref)
+      val t3 = subscribeAndConsumeMessages(consumerClientId3, probe3subscription, topicCount, partitionCount)
+      val (control3, probe3) = (t3._1, t3._2)
+      probe3.ensureSubscription()
+      //probe2.request(topicCount * partitionCount)
+      probe3.request(1)
+      log.debug(s"END:3:Subscribe client ${consumerClientId3} to all topic-partitions in parallel")
+      sleep(rebalanceEventBufferTime, s"SLEEP:3:sleep to allow consume messages by ${consumerClientId3}")
+      // END: introduce third consumer3 with all topic-partitions divided into three consumers
+
+      // BEGIN: cancel consumer1 and assign all topic-partitions to consumer2 and consumer3
+      log.debug(s"BEGIN:1:Cancelling client ${consumerClientId1}")
+      probe1.cancel()
+      control1.shutdown()
+      control1.isShutdown.futureValue shouldBe Done
+      log.debug(s"END:1:Cancelling consumer ${consumerClientId1}")
+      sleep(rebalanceEventBufferTime,
+            s"SLEEP:4:sleep to allow consume messages by ${consumerClientId2} and ${consumerClientId3}")
+      // END: cancel consumer1 and assign all topic-partitions to consumer1 and consumer2
+
+      // BEGIN: cancel consumer2 and assign all topic-partitions to consumer3
+      log.debug(s"BEGIN:2:Cancelling consumer ${consumerClientId2}")
+      probe2.cancel()
+      control2.shutdown()
+      control2.isShutdown.futureValue shouldBe Done
+      log.debug(s"END:2:Cancelling consumer ${consumerClientId2}")
+      sleep(rebalanceEventBufferTime, s"SLEEP:5:sleep to allow consume messages by ${consumerClientId3}")
+      // END: cancel consumer2 and assign all topic-partitions to consumer3
+
+      // BEGIN: Re-subscribe consumer2
+      log.debug(s"BEGIN:4:Re-subscribe client ${consumerClientId2} to all topic-partitions in parallel")
+      probe2rebalanceActor = TestProbe()
+      probe2subscription = Subscriptions.topics(topicSet).withRebalanceListener(probe2rebalanceActor.ref)
+      t2 = subscribeAndConsumeMessages(consumerClientId2, probe2subscription, topicCount, partitionCount)
+      control2 = t2._1; probe2 = t2._2
```

Comment:
What is the purpose of re-using these variables?

Reply:
There isn't much purpose, except to make them follow suffix for client, e.g., control2, probe2 are for client2, whenever that particular client is closed and restarted. I'll rename them.

Reply:
Could mapAsyncUnordered in Committer::batchFlow cause this? Can we make Committer listen on rebalance events and skip commits emitted by defunct consumers?